File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/director_unroll_runme.m
director_unroll

MyFoo=@() subclass(director_unroll.Foo(),'ping',@(self) "MyFoo::ping()");

a = MyFoo();

b = director_unroll.Bar();

b.set(a);
c = b.get();

if (swig_this(a) != swig_this(c))
  a,c
  error
endif

File: support_tools/swigwin-3.0.12/Examples/test-suite/php5/director_unroll_runme.php
<?php

require "tests.php";
require "director_unroll.php";

// No new functions
check::functions(array(foo_ping,foo_pong));
// No new classes
check::classes(array(Foo,Bar));
// now new vars
check::globals(array(bar));

class MyFoo extends Foo {
  function ping() {
    return "MyFoo::ping()";
  }
}

$a = new MyFoo();

$b = new Bar();

$b->set($a);
$c = $b->get();

check::equal($a->this, $c->this, "this failed");

check::done();
?>
